tests/readwhatwaswritten.c

```diff
@@ -252,12 +252,12 @@ int main(int argc, char *argv[]) {
 
     ret = mkdir(dirname, 0755);
     if (ret < 0) {
-        printf("FAIL: Couldn't make directory %s. Exiting\n", dirname);
+        printf("FAIL: Couldn't make directory %s (%i). Exiting\n", dirname, errno);
```

Comment:
are you sure about `%i`?
also I'd add string error message to print  
```
printf("FAIL: Couldn't make directory %s (%d) err=%s. Exiting\n", dirname, errno, strerror(errno));
```

Reply:
%i works for sure (it told me what the error actually was) but I think you're right about strerror(). I'll add it if we think it's worth merging this.

Reply:
Updated.